base/array.jl

```diff
@@ -573,12 +573,11 @@ function collect_to!(dest::AbstractArray{T}, itr, offs, st) where T
     i = offs
     while !done(itr, st)
         el, st = next(itr, st)
-        S = typeof(el)
-        if S === T || S <: T
+        if el isa T || typeof(el) === T
```

Comment:
The need for `typeof(el) === T` is a bit of a mystery, but it was weird to need it in the existing code already. See https://github.com/JuliaLang/julia/issues/25799.

Reply:
I assume that was some attempt at providing a fast-path for uninferred code. But `S === T` is often much harder to infer than `S <: T`. And since `isconcretetype(S)`, it won't be faster at runtime either.

Reply:
Yet if I remove that check I get a worse performance than before with `map(identity, y)` (see what I noted at #25799). That's really weird, but...

Reply:
Keno recently changed the result of `el isa T` from `Const(true)` to `Conditional(:el, T, Union{})` – maybe there's was a mistake in the implementation of that?

Reply:
Where's the relevant commit? Can I try reverting it?